Más ajustes de colecciones y productos

ecc97 opened the pull request:
Esta PR contiene los siguientes cambios

* La data lista para enviar al backend (actualización de data precio para mostrar posteriormente en el listado)
* Listado de productos con el precio mostrado (esperando el ajuste por parte de backend)
- Quitar redirecciones innecesarias, variables inutilizadas, funciones innecesarias, console.log innecesarios
- Botón de eliminar en productos, deshabilitado (no existe flujo en UI)
- Añadir advertencias UI en caso de que no haya materiales registrados al seleccionar material en ficha de prenda
- Ajustar estilos de botones según UI

ecc97 committed:
chore: Adjustments and corrections to collections in product listings and creation, UI adjustments

elianisdev reviewed (approved):
Buen trabajo edwin

kevirui committed:
Merge branch 'develop' of https://github.com/FooTalentGroup/konfex-web-app into feature/collections-settings-flows

kevirui committed:
added coleccion docs

kevirui committed:
update getAll Colecciones

kevirui committed:
update seeder

kevirui committed:
update docs products

kevirui committed:
fix build

ecc97 committed:
Fix/products collections back (#215)

# Nuevas implementaciones en el backend de productos y colecciones.

- Añadí nuevo modelo `colecciones`
- Actualice el modelo de `productos`:
  - Contiene ahora los campos `precio`, `tarifaCosto`, `tarifaHoras`.
- Eliminación de la entidad `mano de obra` y `mano de obra por
producto`.

**LEER DOCUMENTACIÓN EN SWAGGER AL LEVANTAR EN LOCAL**

Nueva respuesta de productos:
```bash
{
  "success": true,
  "statusCode": 200,
  "message": "Productos obtenidos correctamente",
  "data": [
    {
      "id": 1,
      "codigo": 1001,
      "nombre": "Polera Oversize",
      "descripcion": "Polera algodón 240g",
      "activo": true,
      "imagen": "https://ejemplo.com/polera-oversize.jpg",
      "tallas": [
        "S",
        "M",
        "L",
        "XL"
      ],
      "colores": [
        "Negro",
        "Blanco",
        "Gris"
      ],
      "coleccionId": 1,
      "mermaCantidad": 0.5,
      "mermaUnidad": "metros",
      "mermaPrecio": 2.5,
      "tarifaCosto": 15000,
      "tarifaHoras": 2.5,
      "precio": 45000,
      "createdAt": "2024-01-15T10:30:00Z",
      "updatedAt": "2024-01-15T10:30:00Z",
      "coleccion": {
        "id": 1,
        "codigo": 100,
        "nombre": "Colección Verano 2024",
        "imagen": "https://ejemplo.com/coleccion-verano.jpg",
        "icono": "https://ejemplo.com/icono-verano.svg",
        "createdAt": "2024-01-01T00:00:00Z",
        "updatedAt": "2024-01-01T00:00:00Z"
      },
      "materiales": [
        {
          "id": 1,
          "productoId": 1,
          "materialId": 1,
          "cantidad": 1.5,
          "material": {
            "id": 1,
            "nombre": "Tela Jersey Algodón",
            "url_imagen": "https://ejemplo.com/tela-jersey.jpg",
            "categoriaId": 1,
            "categoria": {
              "id": 1,
              "nombre": "Telas",
              "createdAt": "2024-01-01T00:00:00Z",
              "updatedAt": "2024-01-01T00:00:00Z"
            },
            "unidadMedida": "metros",
            "ancho": 1.5,
            "peso": 240,
            "colores": [
              "Negro",
              "Blanco",
              "Gris"
            ],
            "proveedor": "Textiles SA",
            "precio": 8500,
            "createdAt": "2024-01-01T00:00:00Z",
            "updatedAt": "2024-01-01T00:00:00Z"
          }
        },
        {
          "id": 2,
          "productoId": 1,
          "materialId": 2,
          "cantidad": 0.3,
          "material": {
            "id": 2,
            "nombre": "Hilo Poliéster",
            "url_imagen": null,
            "categoriaId": 2,
            "categoria": {
              "id": 2,
              "nombre": "Insumos",
              "createdAt": "2024-01-01T00:00:00Z",
              "updatedAt": "2024-01-01T00:00:00Z"
            },
            "unidadMedida": "metros",
            "ancho": null,
            "peso": null,
            "colores": [
              "Negro",
              "Blanco"
            ],
            "proveedor": "Hilos del Sur",
            "precio": 150,
            "createdAt": "2024-01-01T00:00:00Z",
            "updatedAt": "2024-01-01T00:00:00Z"
          }
        }
      ],
      "pedidos": [],
      "presupuestoDetalles": []
    }
  ]
}
```

Colecciones:
```bash
{
  "success": true,
  "statusCode": 0,
  "message": "string",
  "data": [
    {
      "id": 0,
      "codigo": 0,
      "nombre": "string",
      "imagen": "string",
      "icono": "string",
      "productos": [
        {
          "id": 0,
          "codigo": 0,
          "nombre": "string",
          "descripcion": "string",
          "activo": true,
          "imagen": "string",
          "tallas": [
            "string"
          ],
          "colores": [
            "string"
          ],
          "coleccionId": 0,
          "mermaCantidad": 0,
          "mermaUnidad": "string",
          "mermaPrecio": 0,
          "createdAt": "2025-12-09T22:12:06.435Z",
          "updatedAt": "2025-12-09T22:12:06.435Z"
        }
      ],
      "createdAt": "2025-12-09T22:12:06.435Z",
      "updatedAt": "2025-12-09T22:12:06.435Z"
    }
  ]
}
```

ecc97 committed:
Merge branch 'develop' into feature/collections-settings-flows

kevirui reviewed (changes requested):
Aprobada, verificar si back conecta bien con front y si hay consistencia de datos.

kevirui committed:
fix

kevirui committed:
Merge branch 'feature/collections-settings-flows' of https://github.com/FooTalentGroup/konfex-web-app into feature/collections-settings-flows

kevirui committed:
fix product model

ecc97 committed:
fix(producto): correct data type in repository update method

- Replace Partial<CreateProductDtoDB> with explicit type compatible
- with Prisma's ProductUpdateInput, resolving TS2322 error in coleccionId

ecc97 committed:
fix: correct data types in the garment and product schema
